src/s390_crypto.c

```diff
@@ -467,8 +574,19 @@ int s390_initialize_functionlist()
 		case EC_DSA_SIGN: /* fall-through */
 		case EC_DSA_VERIFY: /* fall-through */
 		case EC_KGEN:
+			if (ecc_via_online_card) {
+				e->flags |= ICA_FLAG_DHW;
+				e->property = 0x0F;
+			}
 			e->flags |= *s390_kdsa_functions[e->id].enabled ? ICA_FLAG_SHW : 0;
 			break;
+		case RSA_ME: /* fall-through */
+		case RSA_CRT:
+			if (any_card_online) {
```

Comment:
I think libica RSA only works with CEX-A, no ? So 'any_card_online' would be neccesary but not sufficient.

Reply:
I found this paragraph in the "Device Drivers, Features, and Commands" book:

> For both accelerators and CCA coprocessors, it provides Rivest-Shamir-Adleman
> (RSA) encryption and RSA decryption functions using clear keys. RSA operations
> are supported in both the modulus-exponent and the Chinese-Remainder Theorem
> (CRT) variants for any key size in the range 57 - 4096 bit.

So it looks like the zcrypt device driver can use both types of cards for RSA.

Reply:
Just tested it: the RSA tests are performed in hw on a CCA coprocessor.

Reply:
Yes, the zcrypt dd can use both.